### DISCUSSION TITLE
Accessing response (or response headers) from caught exception

### DISCUSSION THREAD

**Discussion by ethemguner** [saleweaver/python-amazon-sp-api#258] [Show and tell]
Hi there,

I'm using python-amazon-sp-api for a while to integrate to SP-API. So, there was a problem about rate limit and I needed Request ID and request timestamp to open ticket to Amazon Developer Support. But, when I caught the error, I wasn't able to get response or response headers from exception to get request id & timestamp.

I think It'd be good getting headers from exception. It doesn't has to be response/headers itself though, x-amzn-RequestId and Date would be enough.

Also some of users may want to get rate limit, too.

So, I want to discuss.

I already added headers to exceptions, I can open a pull request.

An example of usage:

```
try:
    response = Orders(
        **order_params
    ).get_order_address(
        order_id=order_id
    )
except SellingApiRequestThrottledException as error:
    # accessing headers:
    error.headers.get("x-amzn-RequestId")
    error.headers.get("Date")
    ...
    # or accessing directly to request id and date (this is not what I did, only a suggestion):
    error.request_id
    error.request_date
    # other codes here to use request_id/date
    ...
```

**Comment by saleweaver** [comment]
Hi, sounds good. Please add a PR! 

Thanks!

**Reply by ethemguner** [reply]
pull request opened, #259
